src/impl/track.cpp

```diff
@@ -182,10 +182,11 @@ bool Track::transportSend([[maybe_unused]] message_ptr message) {
 void Track::setMediaHandler(shared_ptr<MediaHandler> handler) {
 	{
 		std::unique_lock lock(mMutex);
-		mMediaHandler = handler;
+		mMediaHandler = std::move(handler);
 	}
 
-	handler->onOutgoing(std::bind(&Track::transportSend, this, std::placeholders::_1));
+	if (mMediaHandler)
+		mMediaHandler->onOutgoing(std::bind(&Track::transportSend, this, std::placeholders::_1));
```

Comment:
This operates on `mMediaHandler` outside the lock and therefore breaks synchonization. Fixed in https://github.com/paullouisageneau/libdatachannel/pull/369.